orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/UpsertSecurityGroupTask.groovy

```diff
@@ -14,48 +14,40 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.orca.kato.tasks.gce.securitygroup
+package com.netflix.spinnaker.orca.clouddriver.tasks
 
 import com.netflix.spinnaker.orca.DefaultTaskResult
 import com.netflix.spinnaker.orca.ExecutionStatus
-import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.TaskResult
 import com.netflix.spinnaker.orca.clouddriver.KatoService
-import com.netflix.spinnaker.orca.clouddriver.MortService
-import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
-class UpsertGoogleSecurityGroupTask extends AbstractCloudProviderAwareTask implements Task {
+class UpsertSecurityGroupTask extends AbstractCloudProviderAwareTask {
 
   @Autowired
   KatoService kato
 
   @Autowired
-  MortService mortService
+  List<SecurityGroupUpserter> securityGroupUpserters
 
   @Override
   TaskResult execute(Stage stage) {
     String cloudProvider = getCloudProvider(stage)
-    String account = getCredentials(stage)
+    def upserter = securityGroupUpserters.find { it.cloudProvider == cloudProvider }
+    if (!upserter) {
+      throw new IllegalStateException("SecurityGroupUpserter not found for cloudProvider $cloudProvider")
+    }
 
-    def taskId = kato.requestOperations(cloudProvider, [[upsertSecurityGroup: stage.context]])
-                     .toBlocking()
-                     .first()
+    def (ops, extraOutput) = upserter.getOperationsAndExtraOutput(stage)
```

Comment:
I think this might be nicer if it were typed (see earlier comment) or had `getOperations()` and `getExtraOutput` on the interface?